### ISSUE TITLE
Instructor stays on login page after successful login in

### ISSUE THREAD

**Issue by axel1200** [ls1intum/Artemis#2162]
### Describe the bug
Instructors/Admins stay on the login page even after successful login. They can click on other things like "course management" and everything works but they are still shown the login page. Also they can not click on "course overview". The bug goes away if the user is given a course.

Does not apply to students they are shown an empty course overview if they have no courses.

#### To Reproduce

1. Go to create a new Instructor
2.Login as such

#### Expected behavior
Empty course overview is shown 

#### Screenshots
![Screenshot from 2020-10-05 17-09-26](https://user-images.githubusercontent.com/19996713/95097275-93d40300-072d-11eb-9a1c-60a474f4a166.png)
![Screenshot from 2020-10-05 17-09-59](https://user-images.githubusercontent.com/19996713/95097308-a0585b80-072d-11eb-88a1-de01966d2f9b.png)

### Environment

<details><pre>
 - OS: Ubuntu
 - Browser chrome
 - Branch: latest develop
</pre></details>

#### Additional context
Newly setup instance at https://alpro.besec.uni-bonn.de

#### Output in Web Console
Web console stays empty. All network trafics returns 200.

Happy to provide more info.

**Comment by krusche** [comment]
I think this occurs if the user does not have the authority ROLE_USER.
When you create a new user, you have to choose ROLE_USER as well.

As soon as you add the user to a course, Artemis automatically calculates the authorities of the user (because they might have changed).
I agree that this is not obvious, but we inherited this behavior from the official jhipster template, so I guess you can open an issue there as well. The user creation screen is rather a database view and not 100% logical.

I guess all users should always have the authority ROLE_USER expect an administrator explicitly removes it

**Comment by axel1200** [comment]
Yes u are right I could fix it by giving out the user role. But it is not intuative

**Comment by krusche** [comment]
I agree, an easy fix would be if ROLE_USER is preselected when you click on New User

**Comment by axel1200** [comment]
Why would you not show this control to a user?

**Comment by 1-alex98** [comment]
I guess Role Hierarchical Roles should be configured some way right? https://docs.spring.io/spring-security/site/docs/current/reference/html5/#authz-hierarchical-roles

**Comment by 1-alex98** [comment]
If I just guess on how it is done then :
```
@Configuration
public class RoleConfig {
    @Bean
    public RoleHierarchyImpl hierarchy(){
        RoleHierarchyImpl roleHierarchy = new RoleHierarchyImpl();
        roleHierarchy.setHierarchy("""
                ROLE_ADMIN > ROLE_INSTRUCTOR
                ROLE_INSTRUCTOR > ROLE_TUTOR
                ...
                """);
        return roleHierarchy;
    }

    @Bean
    public RoleVoter voter(RoleHierarchyImpl roleHierarchy){
        return new RoleHierarchyVoter(roleHierarchy);
    }
}
```
